sqlglot/optimizer/eliminate_subqueries.py

```diff
@@ -133,6 +133,10 @@ def _eliminate_union(scope, existing_ctes, taken):
 
 
 def _eliminate_derived_table(scope, existing_ctes, taken):
+    # This ensures we don't drop the "pivot" arg from a pivoted subquery
+    if scope.parent.pivots:
+        return None
```

Comment:
An example for why this was needed:

```sql
SELECT * FROM (SELECT x, y FROM tbl) PIVOT (SUM(x) FOR y IN ('a', 'b'))
```

Without the condition I added, the subquery would get eliminated and we'd lose the `pivot` arg because its attached in the `exp.Subquery` node, instead of its `this` arg (which is what the CTE is created of later on).